src/dictionary.rs

```diff
@@ -101,10 +101,12 @@ impl TCFType<CFDictionaryRef> for CFDictionary {
 
 impl CFDictionary {
     pub fn from_CFType_pairs(pairs: &[(CFType, CFType)]) -> CFDictionary {
-        let (keys, values): (Vec<CFTypeRef>,Vec<CFTypeRef>) =
-            pairs.iter()
-            .map(|&(ref key, ref value)| (key.as_CFTypeRef(), value.as_CFTypeRef()))
-            .unzip();
+        let mut keys = Vec::with_capacity(pairs.len());
+        let mut values = Vec::with_capacity(pairs.len());
+        for &(ref key, ref value) in pairs {
+            keys.push(key.as_CFTypeRef());
+            values.push(value.as_CFTypeRef());
+        }
```

Comment:
`unzip` is unfortunately unstable: http://doc.rust-lang.org/std/iter/trait.Iterator.html#tymethod.unzip